File: ios/MyModule.m
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(MyModule, NSObject)

RCT_EXTERN_METHOD(addWithCallback: (NSInteger)x Y: (NSInteger)y Callback: (RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(addWithPromise: (NSInteger)x Y: (NSInteger)y resolver: (RCTPromiseResolveBlock)resolve rejecter: (RCTPromiseRejectBlock)reject)

@end

File: ios/MyModule.swift
import Foundation

@objc(MyModule)
class MyModule: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc
  func addWithCallback(_ x: NSInteger, Y y: NSInteger, Callback callback: RCTResponseSenderBlock) {
//    callback([x + y]) // number
//    callback(["string"]) // string
//    callback([[100, 20, 39]]) // array
    callback([["name": "david", "age": 123 ]]) // object
  }
  
  @objc
  func addWithPromise(_ x: NSInteger, Y y: NSInteger, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
//    resolve(x + y)
    let error = NSError(domain: "", code: 200, userInfo: ["error": "error"])
    reject("code", "messsage", error)
  }
}
